scene/2d/physics_body_2d.cpp

```diff
@@ -1198,7 +1198,7 @@ Vector2 KinematicBody2D::move_and_slide(const Vector2 &p_linear_velocity, const
 						revert_motion();
 						return Vector2();
 					}
-				} else if (get_collision_normal().dot(p_floor_direction) < Math::cos(Math::deg2rad((float)45))) { //ceiling
+				} else if (get_collision_normal().dot(p_floor_direction) <= -Math::cos(Math::deg2rad(p_floor_max_angle))) { //ceiling
```

Comment:
Oh, right. How about we write that the default value is π/4 in the documentation?